File: 782-JewelsAndStones/782-JewelsAndStones.java
// Last updated: 3/23/2026, 5:00:52 PM
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        int count = 0;
        HashMap<Character,Integer> map = new HashMap<>();
        for(char ch : jewels.toCharArray()) map.put(ch,map.getOrDefault(ch,0)+1);
        for(char ch : stones.toCharArray()){
            if(map.containsKey(ch)) count++;
        }
        return count;
    }
}

File: 782-JewelsAndStones/782-JewelsAndStones.cpp
// Last updated: 3/23/2026, 5:00:53 PM
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        int count = 0;
        unordered_map<int,int> hashmap;
        for(char ch : jewels) hashmap[ch]++;
        for(char ch : stones){
            if(hashmap.count(ch)) count++;
        }
        return count;
    }
};